gnocchi/carbonara.py

```diff
@@ -752,12 +758,11 @@ def serialize(self, start, compressed=True):
         offset = int((first - start.key) / offset_div) * self.PADDED_SERIAL_LEN
         return offset, payload
 
-    def _truncate(self, quick=False):
+    def _truncate(self):
         """Truncate the timeserie."""
         if self.max_size is not None:
             # Remove empty points if any that could be added by aggregation
-            self.ts = (self.ts[-self.max_size:] if quick
-                       else self.ts.dropna()[-self.max_size:])
```

Comment:
fun fact: this code is never executed. That's cool because it's wrong: `self.ts.dropna()` does not work anymore since `self.ts` is not a `pandas.Series` anymore.

That means currently `quick` is always `True`. Which is correct since `_truncate` is only called by `__init__` with `quick=True`.